src/file.ts

```diff
@@ -1740,12 +1745,20 @@ class File extends ServiceObject<File> {
       md5,
     });
 
+    const progressStream = new ProgressStream();
+    progressStream.on('progress', bytesRead => {
+      if (options.onUploadProgress) {
+        options.onUploadProgress({bytesRead});
+      }
+    });
+
     const fileWriteStream = duplexify();
 
     const stream = streamEvents(
       pumpify([
         gzip ? zlib.createGzip() : through(),
         validateStream,
+        progressStream,
```

Comment:
I would think this should go after the fileWriteStream to be a representation of the bytes that have been sent to the API.